src/api/middleware/validators/company.js

```diff
@@ -69,12 +69,20 @@ const enable = useExpressValidators([
         .custom(companyExists).withMessage(ValidationReasons.COMPANY_NOT_FOUND),
 ]);
 
+const deleteCompany = useExpressValidators([
+    param("companyId")
+        .exists().withMessage(ValidationReasons.REQUIRED).bail()
+        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
+        .custom(companyExists).withMessage(ValidationReasons.COMPANY_NOT_FOUND),
+]);
+
```

Comment:
I continued with the logic of having one validator per endpoint but this is repeated 3x. Shouldn't we re-utilize one validator for the 3 endpoints?

Reply:
we should have a validator per endpoint to keep the expected structure, what you can do is create an abstraction like this:

```js
const existingCompanyParamValidator = param("companyId")
        .exists().withMessage(ValidationReasons.REQUIRED).bail()
        .custom(isObjectId).withMessage(ValidationReasons.OBJECT_ID).bail()
        .custom(companyExists).withMessage(ValidationReasons.COMPANY_NOT_FOUND)
```

and reuse it wherever it is being used. The advantage is that since you are storing the chain itself, you can add validation after this as well (e.g., `existingCompanyParamValidator.custom(other1).custom(other2)`)

Reply:
I don't know which routes require this validation. If it's only on the company scope, you can create this abstraction on the middleware/company,js, otherwise use the middleware/utils.js

Reply:
I think that would create circular dependencies so I'm going to define it in validators/company.js. You can see how it is later

Reply:
I actually meant validatorUtils, since it will be used by validators. It does not import anything from validators so there shouldn't be any circular dependencies. Nevertheless, if only company validators use it, put it there